app/Http/Controllers/API/VoucherController.php

```diff
@@ -114,8 +121,15 @@ function($dirtycode) {
                 }
                 // If this is a confirm transition - add to a list
                 // for sending to ARC admin. This is a request for payment.
-                if ($transition === 'confirm') {
+                if ($transition === 'confirm' && $stateToken) {
```

Comment:
Do we care if the $stateToken is of type StateToken? Otherwise what is the purpose of the check? Would it be possible to create the token here instead and not need the extra `if === confirm` clause ^^

Reply:
No, this gets fired for every voucher, I'd want one StateToken per batch